### ISSUE TITLE
Better name for the "underlying grid" for `ImmersedBoundaryGrid`

### ISSUE THREAD

**Issue by glwagner** [CliMA/Oceananigans.jl#2095]
Right now `ImmersedBoundaryGrid` has a property `grid`:

https://github.com/CliMA/Oceananigans.jl/blob/a4c27e0f01a003296ff181d4079fa9a5e35b911b/src/ImmersedBoundaries/ImmersedBoundaries.jl#L88-L96

which leads to incredibly silly code like `model.grid.grid`.

Perhaps `ImmersedBoundaryGrid.underlying_grid` or `ImmersedBoundaryGrid.primitive_grid` would be better names.

cc @simone-silvestri

**Comment by francispoulin** [comment]
Good point. Another option is `ImmersedBoundaryGrid.immersed_grid`, but of the other two are also fine choices.

**Comment by glwagner** [comment]
does `immersed_grid` refer to the underlying grid with no boundary, or to the grid that _has_ an immersed boundary?

**Comment by glwagner** [comment]
I guess "immersed grid" could mean "grid with an immersed boundary in it"

**Comment by francispoulin** [comment]
Not sure, so maybe a good reason to not use it.  Hmm....

**Comment by tomchor** [comment]
Is `model.grid.grid` the grid in which the immersed body is? e.g. one that we create with `RectilinearGrid`?

If so, maybe `base_grid`?

**Comment by glwagner** [comment]
I think `base_grid` is a good name. "underlying" is also a little long and "base" helps with that. 

Note we also should change the name of this abstract type if we use a different name than `underlying`:

https://github.com/CliMA/Oceananigans.jl/blob/3c86d8f37a3bf5ff050e233f7b946685f9057c26/src/Grids/Grids.jl#L86-L92

Another possibility that comes to mind now is `parent_grid`, which is used throughout Julia to indicate a "wrapped" object:

```julia
```julia
help?> parent
search: parent parentmodule parentindices

  parent(A)

  Return the underlying "parent array”. This parent array of objects of types SubArray, ReshapedArray or LinearAlgebra.Transpose is what was passed as an argument to view, reshape, transpose, etc. during object creation. If the input is not a wrapped object, return the input itself.

  Examples
  ≡≡≡≡≡≡≡≡≡≡

  julia> A = [1 2; 3 4]
  2×2 Matrix{Int64}:
   1  2
   3  4

  julia> V = view(A, 1:2, :)
  2×2 view(::Matrix{Int64}, 1:2, :) with eltype Int64:
   1  2
   3  4

  julia> parent(V)
  2×2 Matrix{Int64}:
   1  2
   3  4
```

(And there's also `parent module`, etc.) Perhaps that docstring is partly inspiration for referring to things as "underlying" (elsewhere we also have descriptions of "underlying data" for fields.

But I like either `parent` or `base` better than `underlying`, with maybe a preference for `parent` now that I see its fairly widespread.

As a side note (issue creep...) we also don't need `AbstractRectilinearGrid`, since there's only one:

https://github.com/CliMA/Oceananigans.jl/blob/3c86d8f37a3bf5ff050e233f7b946685f9057c26/src/Grids/rectilinear_grid.jl#L8

eh?

**Comment by francispoulin** [comment]
I like `base.grid`

**Comment by glwagner** [comment]
we call it `underlying_grid`
